{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3e3c92aa-3846-438d-9cde-2ac8425ba654",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "STEP1\n",
      "usage: magpurify [-h] [--version]\n",
      "                 {phylo-markers,clade-markers,conspecific,tetra-freq,gc-content,coverage,known-contam,clean-bin}\n",
      "                 ...\n",
      "\n",
      "Identify and remove incorrectly binned contigs from metagenome-assembled\n",
      "genomes.\n",
      "\n",
      "positional arguments:\n",
      "  {phylo-markers,clade-markers,conspecific,tetra-freq,gc-content,coverage,known-contam,clean-bin}\n",
      "    phylo-markers       find taxonomic discordant contigs using a database of\n",
      "                        phylogenetic marker genes.\n",
      "    clade-markers       find taxonomic discordant contigs using a database of\n",
      "                        clade-specific marker genes.\n",
      "    conspecific         find contigs that fail to align to closely related\n",
      "                        genomes.\n",
      "    tetra-freq          find contigs with outlier tetranucleotide frequency.\n",
      "    gc-content          find contigs with outlier GC content.\n",
      "    coverage            find contigs with outlier coverage profile.\n",
      "    known-contam        find contigs that match a database of known\n",
      "                        contaminants.\n",
      "    clean-bin           remove putative contaminant contigs from bin.\n",
      "\n",
      "optional arguments:\n",
      "  -h, --help            show this help message and exit\n",
      "  --version             show program's version number and exit\n",
      "STEP2\n",
      "usage: magpurify [-h] [--version]\n",
      "                 {phylo-markers,clade-markers,conspecific,tetra-freq,gc-content,coverage,known-contam,clean-bin}\n",
      "                 ...\n",
      "\n",
      "Identify and remove incorrectly binned contigs from metagenome-assembled\n",
      "genomes.\n",
      "\n",
      "positional arguments:\n",
      "  {phylo-markers,clade-markers,conspecific,tetra-freq,gc-content,coverage,known-contam,clean-bin}\n",
      "    phylo-markers       find taxonomic discordant contigs using a database of\n",
      "                        phylogenetic marker genes.\n",
      "    clade-markers       find taxonomic discordant contigs using a database of\n",
      "                        clade-specific marker genes.\n",
      "    conspecific         find contigs that fail to align to closely related\n",
      "                        genomes.\n",
      "    tetra-freq          find contigs with outlier tetranucleotide frequency.\n",
      "    gc-content          find contigs with outlier GC content.\n",
      "    coverage            find contigs with outlier coverage profile.\n",
      "    known-contam        find contigs that match a database of known\n",
      "                        contaminants.\n",
      "    clean-bin           remove putative contaminant contigs from bin.\n",
      "\n",
      "optional arguments:\n",
      "  -h, --help            show this help message and exit\n",
      "  --version             show program's version number and exit\n",
      "STEP3\n",
      "usage: magpurify [-h] [--version]\n",
      "                 {phylo-markers,clade-markers,conspecific,tetra-freq,gc-content,coverage,known-contam,clean-bin}\n",
      "                 ...\n",
      "\n",
      "Identify and remove incorrectly binned contigs from metagenome-assembled\n",
      "genomes.\n",
      "\n",
      "positional arguments:\n",
      "  {phylo-markers,clade-markers,conspecific,tetra-freq,gc-content,coverage,known-contam,clean-bin}\n",
      "    phylo-markers       find taxonomic discordant contigs using a database of\n",
      "                        phylogenetic marker genes.\n",
      "    clade-markers       find taxonomic discordant contigs using a database of\n",
      "                        clade-specific marker genes.\n",
      "    conspecific         find contigs that fail to align to closely related\n",
      "                        genomes.\n",
      "    tetra-freq          find contigs with outlier tetranucleotide frequency.\n",
      "    gc-content          find contigs with outlier GC content.\n",
      "    coverage            find contigs with outlier coverage profile.\n",
      "    known-contam        find contigs that match a database of known\n",
      "                        contaminants.\n",
      "    clean-bin           remove putative contaminant contigs from bin.\n",
      "\n",
      "optional arguments:\n",
      "  -h, --help            show this help message and exit\n",
      "  --version             show program's version number and exit\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "# -> 切換目錄\n",
    "cd ${HOME}/ncku_slurm\n",
    "\n",
    "# -> 環境變數\n",
    "export MAGPURIFYDB=${HOME}/MAGpurify-db-v1.0\n",
    "\n",
    "# -> 執行程式\n",
    "ml libs/singularity/3.7.1\n",
    "\n",
    "echo \"STEP1\"\n",
    "singularity exec \\\n",
    "    /work/c00cjz002/nvidia/magpurify_v2.1.2.sif bash -c \\\n",
    "    \"magpurify -h && sleep 5 && echo $(date) > date.txt\"\n",
    "\n",
    "echo \"STEP2\"\n",
    "singularity exec \\\n",
    "    /work/c00cjz002/nvidia/magpurify_v2.1.2.sif bash -c \\\n",
    "    \"magpurify -h && sleep 5 && echo $(date) >> date.txt\" \n",
    "\n",
    "echo \"STEP3\"\n",
    "singularity exec \\\n",
    "    /work/c00cjz002/nvidia/magpurify_v2.1.2.sif bash -c \\\n",
    "    \"magpurify -h && sleep 5 && echo $(date) >> date.txt\" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "75c2ffc3-3d11-4509-8fb5-db911750805d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
